src/tiny_slam/tiny_scan_matcher.h

```diff
@@ -6,16 +6,36 @@
 #include "../core/monte_carlo_scan_matcher.h"
 #include "tiny_scan_cost_estimator.h"
 
+/*!
+ * \brief The scan matcher based on Monte Carlo simulation.
+ *
+ * The simulation of the robot pose by shifting it on a random vector and
+ * rotation by a random angle. The vector distribution is dynamically adjusted.
+ */
 class TinyScanMatcher : public MonteCarloScanMatcher {
 private:
   using ScePtr = std::shared_ptr<ScanCostEstimator>;
 public:
+  /*!
+   * Initializes the scan matcher.
```

Comment:
I think the distinction between bad_iter and max_iter params is not clear enough.